PDF Plugin should load its inline styles regardless of CSP

stwrt opened the pull request:
#### 118097374f64baa854137fa9b507f83270ff4b67
<pre>
PDF Plugin should load its inline styles regardless of CSP
<a href="https://bugs.webkit.org/show_bug.cgi?id=272689">https://bugs.webkit.org/show_bug.cgi?id=272689</a>
<a href="https://rdar.apple.com/125796665">rdar://125796665</a>

Reviewed by Alex Christensen.

We embed styles when rendering a pdf file that should not be ignored if CSP is activated (particularly around style sheets),
and will lead to a poor viewing experience for the user.

In addition, Anne pointed out that CSP is not defined for PDF documents, and thus we should ignore the checks.

* LayoutTests/pdf/fullscreen-expected.txt: Added.
* LayoutTests/pdf/fullscreen.html: Added.
* LayoutTests/pdf/resources/fullscreen.pdf: Added.
* LayoutTests/platform/gtk/pdf/fullscreen-expected.txt: Added.
* LayoutTests/platform/ios/pdf/fullscreen-expected.txt: Added.
* LayoutTests/platform/mac-wk1/pdf/fullscreen-expected.txt: Added.
* LayoutTests/platform/wpe/pdf/fullscreen-expected.txt: Added.
* Source/WebCore/dom/InlineStyleSheetOwner.cpp:
(WebCore::InlineStyleSheetOwner::createSheet):

Canonical link: <a href="https://commits.webkit.org/277683@main">https://commits.webkit.org/277683@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/abe65162f286c8b22d5576b41b7f58e1bce56a54

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48274 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27486 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51228 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50962 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44339 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50579 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33420 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25006 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39435 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48856 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25180 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/41706 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20583 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22660 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/42886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6330 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44618 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43341 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52866 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23321 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19673 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46779 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24586 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41889 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45689 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10654 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25391 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24309 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on LayoutTests/pdf/fullscreen.html:
```diff
@@ -0,0 +1,16 @@
+<script>
+    if (window.testRunner) {
+        testRunner.dumpAsText();
+        testRunner.notifyDone();
```
You don't need `notifyDone()` if you never called `waitUntilDone()` but does the test need to wait for the PDF to load?

stwrt replied:
Yes, we need to wait for this test case to finish.

I'll need to fix this.

stwrt replied:
Updated the test case to wait for the page to finish loading.

smfr commented on Source/WebCore/dom/InlineStyleSheetOwner.cpp:
```diff
@@ -146,7 +147,7 @@ void InlineStyleSheetOwner::createSheet(Element& element, const String& text)
 
     ASSERT(document->contentSecurityPolicy());
-    if (!document->checkedContentSecurityPolicy()->allowInlineStyle(document->url().string(), m_startTextPosition.m_line, text, CheckUnsafeHashes::No, element, element.nonce(), element.isInUserAgentShadowTree())) {
+    if (!document->checkedContentSecurityPolicy()->allowInlineStyle(document->url().string(), m_startTextPosition.m_line, text, CheckUnsafeHashes::No, element, element.nonce(), element.isInUserAgentShadowTree() || dynamicDowncast<PluginDocument>(document))) {
```
Just use `is<PluginDocument>()`

stwrt replied:
Thanks for the correction.

stwrt committed:
PDF Plugin should load its inline styles regardless of CSP
https://bugs.webkit.org/show_bug.cgi?id=272689
rdar://125796665

Reviewed by Alex Christensen.

We embed styles when rendering a pdf file that should not be ignored if CSP is activated (particularly around style sheets),
and will lead to a poor viewing experience for the user.

In addition, Anne pointed out that CSP is not defined for PDF documents, and thus we should ignore the checks.

* LayoutTests/pdf/fullscreen-expected.txt: Added.
* LayoutTests/pdf/fullscreen.html: Added.
* LayoutTests/pdf/resources/fullscreen.pdf: Added.
* LayoutTests/platform/gtk/pdf/fullscreen-expected.txt: Added.
* LayoutTests/platform/ios/pdf/fullscreen-expected.txt: Added.
* LayoutTests/platform/mac-wk1/pdf/fullscreen-expected.txt: Added.
* LayoutTests/platform/wpe/pdf/fullscreen-expected.txt: Added.
* Source/WebCore/dom/InlineStyleSheetOwner.cpp:
(WebCore::InlineStyleSheetOwner::createSheet):

Canonical link: https://commits.webkit.org/277683@main
